bundles/binding/org.openhab.binding.panstamp/src/main/java/org/openhab/binding/panstamp/internal/PanStampBinding.java

```diff
@@ -0,0 +1,583 @@
+					stopDebug();
+					startDebug(newCfg);
+				}
+			}
+			cfg = newCfg;
+		} catch (ValueException e) {
+			logger.error(e.getMessage());
+		}
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	public void deactivate(final int reason) {
+		// deallocate resources here that are no longer needed and
+		// should be reset when activating this binding again
+		logger.debug("deactivate()");
+		stop();
+		super.deactivate();
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected String getName() {
+		return "panStamp binding";
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		logger.debug("internalReceiveCommand({},{})", itemName, command);
+		List<PanStampBindingConfig<?>> confs = getConfigs(itemName);
+		for (PanStampBindingConfig<?> conf : confs) {
+			try {
+				PanStamp dev;
+				if (!network.hasDevice(conf.getAddress())) {
+					dev = new PanStamp(network, conf.getAddress());
+					network.addDevice(dev);
+				} else {
+					dev = network.getDevice(conf.getAddress());
+				}
+				if (dev.hasRegister(conf.getRegister())) {
+					Register reg = dev.getRegister(conf.getRegister());
+					if (reg.hasEndpoint(conf.getEndpoint())) {
+						Endpoint<Object> ep = reg.getEndpoint(conf
+								.getEndpoint());
+						if (ep.isOutput()) {
+							Object val = null;
+							switch (ep.getType()) {
+							case BINARY:
+								val = PanStampConversions.toBoolean(command);
+								break;
+							case INTEGER:
+								val = PanStampConversions.toInteger(command);
+								break;
+							case NUMBER:
+								val = PanStampConversions.toDouble(command);
+								break;
+							case STRING:
+								val = PanStampConversions.toString(command);
+								break;
+							default:
+								logger.error(
+										"Unsupported panStamp endpoint type '{}' for endpoint {}. BUG!",
+										ep.getType(),
+										PanStampConversions.toString(ep));
+							}
+							if (val != null) {
+								if (!conf.getUnit().equals("")) {
+									ep.setValue(conf.getUnit(), val);
+								} else {
+									ep.setValue(val);
+								}
+							}
+						} else {
+							logger.error(
+									"Internal update received for input-only endpoint {}",
+									PanStampConversions.toString(ep));
+						}
+					} else {
+						logger.error(
+								"No endpoint '{}' found for register '{}' on panStamp '{}' linked to item '{}'",
+								conf.getEndpoint(), conf.getRegister(),
+								conf.getAddress(), itemName);
+					}
+				} else {
+					logger.error(
+							"No register '{}' found on panStamp '{}' linked to item '{}'",
+							conf.getRegister(), conf.getAddress(), itemName);
+				}
+			} catch (EndpointNotFoundException e) {
+				logger.error(e.getMessage());
+			} catch (NetworkException e) {
+				logger.error(e.getMessage());
+			} catch (ValueException e) {
+				logger.error(e.getMessage());
+			}
+		}
+	}
+
+	@Override
+	protected void execute() {
```

Comment:
If you don't use the polling of an AbstractActiveBinding, extend AbstractBinding instead.